### ISSUE TITLE
Make coala a GitHub App

### ISSUE THREAD

**Issue by li-boxuan** [coala/projects#699]
See https://github.com/marketplace/category/code-quality

I guess coala would be much more popular if it can be made a GitHub App.

This means coala needs to be run as a web service, deployed somewhere like Heroku (but it's not totally free hmmm....)

GitMate + coala actually do the same thing, but that means developers need to learn both GitMate and coala, and GitMate requires admin access, which sounds a bit risky to developers (that's because GitMate has other functionalities, but coala does not need that high-level permission).

**Comment by budukhyash** [comment]
@li-boxuan Sir I would love to work on this project.

**Comment by li-boxuan** [comment]
@budukhyash Consider writing your ideas/thoughts on this and ask people to review/discuss, so we have a better idea of whether this is viable and worthwhile.

**Comment by budukhyash** [comment]
@li-boxuan I searched about github apps and integrations.I found out a nodejs based framework called as [probot](https://probot.github.io/).I went through the probot docs and created a test github app. I have a few doubts regarding the coala app.

- Gitmate does provide static code analysis and other functionalities and due to which it needs high level permissions.So we are looking to develop a coala app similar to the gitmate in terms of code analysis
which does not require high level permissions.

- coala is interactive in nature.So should the coala app must take input from the users in the form of comments and apply the patch ?

**Comment by budukhyash** [comment]
@jayvdb Hello sir can you please review this.

**Comment by ksdme** [comment]
Related to https://github.com/coala/projects/issues/702

@li-boxuan IIRC we have a docker based coala image, building one click installers for a couple of popular cloud services (Heroku, AWS, GC) would be a good first step.

**Comment by jayvdb** [comment]
> GitMate requires admin access, which sounds a bit risky to developers ...

This is an interesting problem, and should be solvable within GitMate, allowing a non-admin install into an org / repo.

Disabling GitMate features/plugins which are not possible based on the permissions given to GitMate would be in scope.

Another direction to follow is GitMate build process emitting a 'coala-only' mode installer.

Also we do have a hosted version of coala inside the deployed coala webservices (fully functional, exposed via an API) , and inside the deployed corobo (not exposed).  Either of those could be the basis of a github app, but both require some work to get operational again.

**Comment by frextrite** [comment]
> Also we do have a hosted version of coala inside the deployed coala webservices (fully functional, exposed via an API) 

Which deployed web service are we talking about here?

> Another direction to follow is GitMate build process emitting a 'coala-only' mode installer.

What is meant by 'coala-only' mode? Could you explain this a bit more?

**Comment by sladyn98** [comment]
@frextrite https://webservices.coala.io/ have a look here.
@jayvdb 

> Another direction to follow is GitMate build process emitting a 'coala-only' mode installer.

Are we talking about a coala type gitmate clone, because we could make a separate gitmate-style coala app, which would not require high level permissions and deploy it on a webservice.The API's for the webservice could be used as a start.We could follow a MVC based django model.

**Comment by AkshJain99** [comment]
@li-boxuan are you going to mentor this project this year?

**Comment by li-boxuan** [comment]
@AkshJain99 Nope, and I am not even sure if this is suitable for a GSoC project. I open up the issue just because I find it's an interesting open problem.

**Comment by AkshJain99** [comment]
okay @li-boxuan
